addons/web/static/src/search/search_arch_parser.js

```diff
@@ -159,9 +159,21 @@ export class SearchArchParser extends XMLParser {
                     preField.defaultAutocompleteValue.label = option[1];
                 } else if (fieldType === "many2one") {
                     this.labels.push((orm) => {
-                        return orm.call(relation, "name_get", [value], { context }).then((results) => {
-                            preField.defaultAutocompleteValue.label = results[0][1];
-                        });
+                        return orm
+                            .call(relation, "name_get", [value], { context })
+                            .then((results) => {
+                                preField.defaultAutocompleteValue.label = results[0][1];
+                            });
+                    });
+                } else if (["many2many", "one2many"].includes(fieldType) && Array.isArray(val) && val.every((v) => Number.isInteger(v) && v >0)) {
+                    preField.defaultAutocompleteValue.operator = "in";
```

Comment:
is it what we want.
something I don't like: we have the same facet texts when a X2many is activated through search_default_ or the search bar but with different resulting domain. If id=1 has name "Marc", we get via search_default_user_ids: [1],
[("user_ids", in, [1])] and facet text Users: Marc
while type Marc in the search bar and select Users gives [("user_ids", "ilike", "Marc")] and facet text Users: Marc.

Solution here is maybe not good -> IMP in master?